presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -364,18 +376,19 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
         return new PrestoSparkTaskSourceRdd(sparkContext.sc(), taskSourcesByPartitionId);
     }
 
-    private List<ScheduledSplit> getSplits(Session session, TableScanNode tableScan)
+    private List<ScheduledSplit> getSplits(TableScanNode tableScan, Map<PlanNodeId, SplitSource> splitSources)
     {
         List<ScheduledSplit> splits = new ArrayList<>();
-        SplitSource splitSource = splitManager.getSplits(session, tableScan.getTable(), UNGROUPED_SCHEDULING);
-        long sequenceId = 0;
-        while (!splitSource.isFinished()) {
-            List<Split> splitBatch = getFutureValue(splitSource.getNextBatch(NOT_PARTITIONED, Lifespan.taskWide(), 1000)).getSplits();
-            for (Split split : splitBatch) {
-                splits.add(new ScheduledSplit(sequenceId++, tableScan.getId(), split));
+        try (SplitSource splitSource = requireNonNull(splitSources.get(tableScan.getId()), "split source is missing for table scan node with id: " + tableScan.getId())) {
```

Comment:
`CloseableSplitSourceProvider` will close all `splitSource`, this try statement is not needed.

Reply:
Yeah, eventually everything is going to get closed by the `CloseableSplitSourceProvider`, but I thought it might be better to close the split sources as soon as possible to decrease memory pressure / etc.

Reply:
Thanks for explanation.